rust/noosphere-gateway/src/handlers/v0alpha2/push.rs

```diff
@@ -149,7 +153,8 @@ where
             return Err(PushError::UnexpectedBody);
         };
 
-        let gateway_sphere_tip = self.sphere_context.version().await?;
+        let gateway_sphere_context = self.gateway_sphere.sphere_context().await?;
+        let gateway_sphere_tip = gateway_sphere_context.version().await?;
```

Comment:
Why not `self.gateway_sphere.version()`?

Reply:
The proceeding `sphere_context()` locks the mutex (hanging the `gateway_sphere.version()` call) -- getting the version from the `gateway_sphere` first would just incur an extra lock, IIUC